data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/PipelinesWithAcksIT.java

```diff
@@ -0,0 +1,207 @@
+                .untilAsserted(() -> {
+            List<Record<Event>> outputRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+            assertThat(outputRecords, not(empty()));
+            assertThat(outputRecords.size(), equalTo(2*numRecords));
+        });
+        assertTrue(inMemorySourceAccessor.getAckReceived());
+    }
+
+    @Test
+    void three_pipelines_multi_sink_multiple_records() {
+        setUp(THREE_PIPELINES_MULTI_SINK_CONFIGURATION_UNDER_TEST);
+        final int numRecords = 100;
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER, numRecords);
+
+        await().atMost(10000, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> {
+            List<Record<Event>> outputRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+            assertThat(outputRecords, not(empty()));
+            assertThat(outputRecords.size(), equalTo(3*numRecords));
+        });
+        assertTrue(inMemorySourceAccessor.getAckReceived());
+    }
+
+    @Test
+    void one_pipeline_three_sinks_multiple_records() {
+        setUp(ONE_PIPELINE_THREE_SINKS_CONFIGURATION_UNDER_TEST);
+        final int numRecords = 100;
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER, numRecords);
+
+        await().atMost(10000, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> {
+            List<Record<Event>> outputRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+            assertThat(outputRecords, not(empty()));
+            assertThat(outputRecords.size(), equalTo(3*numRecords));
+        });
+        assertTrue(inMemorySourceAccessor.getAckReceived());
+    }
+
+    @Test
+    void one_pipeline_ack_expiry_multiple_records() {
+        setUp(ONE_PIPELINE_ACK_EXPIRY_CONFIGURATION_UNDER_TEST);
+        final int numRecords = 100;
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER, numRecords);
+
+        await().atMost(10000, TimeUnit.MILLISECONDS)
+          .untilAsserted(() -> {
+            List<Record<Event>> outputRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+            assertThat(outputRecords, not(empty()));
+            assertThat(outputRecords.size(), equalTo(numRecords));
+        });
+        assertThat(inMemorySourceAccessor.getAckReceived(), equalTo(null));
+    }
+
+    @Test
+    void one_pipeline_three_sinks_negative_ack_multiple_records() {
+        setUp(ONE_PIPELINE_THREE_SINKS_CONFIGURATION_UNDER_TEST);
+        final int numRecords = 100;
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER, numRecords);
+        inMemorySinkAccessor.setResult(false);
+
+        await().atMost(10000, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> {
+            List<Record<Event>> outputRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+            assertThat(outputRecords, not(empty()));
+            assertThat(outputRecords.size(), equalTo(3*numRecords));
+        });
+        assertTrue(!inMemorySourceAccessor.getAckReceived());
```

Comment:
Please use `assertFalse`. Using a `!` is quite subtle and can easily be overlooked.